use/6/4/b.txt

```diff
@@ -0,0 +1,18 @@
+# Use 6.4 Running workflows
+
+## Requirements:
+
+* [[skill-tree:use:5:1:3:b]]
+* [[skill-tree:use:4:2:b]]
+* optionally: [[skill-tree:sd:3:1:b]]
+
+## Learning Objectivs
+
+- Understanding Job Scheduling Systems: Learners will identify "their" job scheduling systems (e.g. SLURM, LSF, HT-Torque) to manage and execute workflows.
+- Submitting Workflows to a Cluster: Learners will be able to submit workflows to HPC systems. For Snakemake this requires providing the appropriate plugins.
+- Monitoring Workflow Execution: Learners will utilize job monitoring tools (e.g. `squeue`/`sacct` for SLURM, `top`,`htop` in case of host based ssh-logins to compute nodes) to track the status and performance of running workflows on HPC systems, identifying potential issues or bottlenecks.
+- Handling Job Failures: 
+  - Learners will diagnose and troubleshoot common errors and failures encountered during workflow execution, implementing strategies to rectify issues and re-run workflows as needed.
+  - Learners will interpret workflow logs to assess the execution process, identify issues, and validate the correctness of results generated by the workflow.
+  - Learners are encouraged to report issues, thereby differentiating between programm failures (i.e. due to bugs), issues of the workflow, the workflow manager and the HPC system (e.g file system or node failures).
+- Collecting and Managing Output Data: Learners will be able to instruct their Workflow Management System to collect output data generated during execution and generate publication ready reports.
```

Comment:
Identify HPC job scheduling systems (e.g. SLURM, LSF, HT-Torque) and describe their role in managing and executing workflows.

Submit workflows to HPC systems, applying the appropriate Snakemake plugins.

Utilise job monitoring tools (e.g. squeue/sacct for SLURM, top, htop) to track the status and performance of running workflows, identify potential issues or bottlenecks.

Diagnose and troubleshoot common errors and failures encountered during workflow execution, implementing strategies to rectify issues and rerun workflows as needed.

Interpret workflow logs to assess the execution process, identify issues, and validate the correctness of generated results.

Report and differentiate between programme failures (due to bugs), workflow or workflow manager issues, and HPC system-level problems (e.g. file system or node failures).

Collect and manage output data generated during execution, instructing the Workflow Management System to produce publication-ready reports.